spec.html

```diff
@@ -34532,8 +34534,203 @@ <h2>Syntax</h2>
           [+UnicodeMode] `-`
           CharacterClassEscape[?UnicodeMode]
           CharacterEscape[?UnicodeMode]
+
+        ClassSetExpression ::
+          ClassUnion
+          ClassIntersection
+          ClassSubtraction
+
+        ClassUnion ::
+          ClassSetRange ClassUnion?
+          ClassSetOperand ClassUnion?
+
+        ClassIntersection ::
+          ClassSetOperand `&amp;&amp;` [lookahead != `&amp;`] ClassSetOperand
+          ClassIntersection `&amp;&amp;` [lookahead != `&amp;`] ClassSetOperand
+
+        ClassSubtraction ::
+          ClassSetOperand `--` ClassSetOperand
+          ClassSubtraction `--` ClassSetOperand
+
+        ClassSetRange ::
+          ClassSetCharacter `-` ClassSetCharacter
+      </emu-grammar>
+      <ul>
+        <li>
+          It is a Syntax Error if the CharacterValue of the first |ClassSetCharacter| is larger than the CharacterValue of the second |ClassSetCharacter|.
+        </li>
+      </ul>
+      <emu-grammar type="definition">
+        ClassSetOperand ::
+          ClassSetCharacter
+          ClassStringDisjunction
+          NestedClass
+
+        NestedClass ::
+          `[` [lookahead != `^`] ClassContents[+UnicodeMode, +UnicodeSetsMode] `]`
+          `[` `^` ClassContents[+UnicodeMode, +UnicodeSetsMode] `]`
+          `\` CharacterClassEscape[+UnicodeMode]
       </emu-grammar>
+      <emu-note>
+        <p>The first two lines here are equivalent to CharacterClass.</p>
+        <p style="background: yellow">[It is less confusing not having to back out of the new syntax, only to dive back in. And if we wanted to use CharacterClass here, we would need to check for differences in static &amp; runtime semantics.]</p>
+      </emu-note>
+      <emu-grammar type="definition">
+        ClassStringDisjunction ::
+          `\q{` ClassString MoreClassStrings? `}`
+
+        MoreClassStrings ::
+          `|` ClassString MoreClassStrings?
+
+        ClassString ::
+          [empty]
+          NonEmptyClassString
+
+        NonEmptyClassString ::
+          ClassSetCharacter NonEmptyClassString?
 
+        ClassSetCharacter ::
+          [lookahead &notin; ClassSetReservedDoublePunctuator] SourceCharacter but not ClassSetSyntaxCharacter
+          `\` CharacterEscape[+UnicodeMode]
+          `\` ClassSetReservedPunctuator
+          `\` `b`
+
+        ClassSetReservedDoublePunctuator :: one of
+          `&amp;&amp;` `!!` `##`
+          `$$` `%%` `**`
+          `++` `,,` `..`
+          `::` `;;` `&lt;&lt;`
+          `==` `&gt;&gt;` `??`
+          `@@` `^^` `__`
```

Comment:
```suggestion
          `@@` `^^`
```

Reply:
Moved to https://github.com/mathiasbynens/ecma262/pull/12